kore/src/Kore/Step/Step.hs

```diff
@@ -397,19 +397,42 @@ finalizeRulesParallel
     => Pattern (Target variable)
     -> [UnifiedRule (Target variable)]
     -> unifier (Results variable)
-finalizeRulesParallel initial unifiedRules = do
-    let initialCondition = Conditional.withoutTerm initial
-    results <-
-        Foldable.fold <$> traverse (finalizeRule initialCondition) unifiedRules
-    let unifications = MultiOr.make (Conditional.withoutTerm <$> unifiedRules)
-        remainder = Predicate.fromPredicate (Remainder.remainder' unifications)
-    remainders' <- Monad.Unify.gather $ applyRemainder initial remainder
-    return Step.Results
-        { results = Seq.fromList results
-        , remainders =
-            OrPattern.fromPatterns
-            $ Pattern.mapVariables Target.unwrapVariable <$> remainders'
-        }
+finalizeRulesParallel initial unifiedRules =
+    case Rule.checkFunctionLike (term <$> unifiedRules) (term initial) of
+        Left err -> error $ "Rules:\n" <> show (unparseToString . TermLike.mapVariables Target.unwrapVariable . Rule.left . term <$> unifiedRules) <> err
```

Comment:
Because this case only throws an error, it would be better if the `Right` case did `return ()`  and the rest of the function followed at the previous indentation level.